src/Microsoft.Azure.WebJobs.Logging.ApplicationInsights/TokenCredentialOptions.cs

```diff
@@ -0,0 +1,123 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using Azure.Core;
+using Azure.Identity;
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.Azure.WebJobs.Logging.ApplicationInsights
+{
+    public class TokenCredentialOptions
+    {
+        private const string AuthToken = "AAD";
+        private const string AuthAuthorizationKey = "Authorization";
+        private const string AuthClientIdKey = "ClientId";
+
+        public TokenCredentialOptions(string authenticationString)
+        {
+            if (!string.IsNullOrWhiteSpace(authenticationString))
+            {
+                var tokens = ParseAuthenticationString(authenticationString);
+
+                if (tokens.ContainsKey(AuthAuthorizationKey))
+                {
+                    Authorization = tokens[AuthAuthorizationKey];
+                }
+                if (tokens.ContainsKey(AuthClientIdKey))
+                {
+                    ClientId = tokens[AuthClientIdKey];
+                }
+            }
+        }
+        /// <summary>
+        /// The authentication kind.
+        /// </summary>
+        public string Authorization { get; set; }
```

Comment:
This might not be necessary to store (for now). We only support a value of `AAD` correct? We can validate that during connection string parsing and throw if it is not `AAD`. Failing fast here will help expose bad configuration earlier to the developer.